projects/discussion-ui/src/lib/components/discuss-category/discuss-category.component.ts

```diff
@@ -107,15 +111,22 @@ export class DiscussCategoryComponent implements OnInit, OnDestroy {
       this.categoryId = _.get(response, 'cid');
       this.isTopicCreator = _.get(response, 'privileges.topics:create') === true ? true : false;
       this.showStartDiscussionModal = false;
+      let input
       if (_.get(response, 'children').length > 0) {
-        this.router.navigate([], { relativeTo: this.activatedRoute.parent, queryParams: { cid: this.categoryId } });
-        this.categories = [];
+        input = { data: { url: '', queryParams: { cid: this.categoryId } }, action: this.categoryAction}
+        this.navigationService.navigate(input)
+
         _.get(response, 'children').forEach(subCategoryData => {
           this.categories.push(subCategoryData);
         });
       } else {
         this.discussService.setContext(CONTEXT_PROPS.cid, this.categoryId);
-        this.router.navigate([`${this.configService.getRouterSlug()}${CONSTANTS.ROUTES.CATEGORY}`, `${this.categoryId}`]);
+        this.configService.setCategoryid(this.categoryId)
+
+        let routerSlug = this.configService.getConfig().routerSlug ? this.configService.getConfig().routerSlug : ''
+        input = { data: { url: `${routerSlug}${CONSTANTS.ROUTES.CATEGORY}${this.categoryId}`, queryParams: {} }, action: CONSTANTS.CATEGORY_HOME, }
+        this.navigationService.navigate(input)
+        this.stateChange.emit({ action: this.categoryAction, categoryId: this.categoryId })
```

Comment:
instead of passing the only categoryID in the output. it's better to pass the complete category object what you get in response.

Reply:
categoryId is the complete response we have there.